src/agent/src/uevent.rs

```diff
@@ -22,6 +24,30 @@ struct Uevent {
     interface: String,
 }
 
+#[cfg(target_arch = "aarch64")]
+pub static mut PCI_ROOT_BUS_PATH: String = String::new();
```

Comment:
I'm not a rust expert (I'm still learning) but I'd use [`lazy_static`][1] to initialize this variable, maybe I'm wrong

cc @jodh-intel @lifupan 

[1]: https://docs.rs/lazy_static/1.4.0/lazy_static/

Reply:
I'm not sure if we can initialize a mutable static variable using lazy_static. compiler will complain if this initialization code put into the lazy_static! block, like:
`lazy_static!{
        pub static mut PCI_ROOT_BUS_PATH: String = String::new();
}
`
Also I find [lazy_static](https://sodocumentation.net/rust/topic/1244/globals#lazy-static-). It says that lazy_static is used for creating immutable variable.

Reply:
Why not  define a get_pci_root_bus fn for Aarch64 and other platfrom, thus you can use the same code to deal with the  PCI_ROOT_BUS_PATH for different platform and avoided those unsafe codes?

Reply:
I'm not clear with that. these unsafe code derived from the mutable static variable not the arch-related code. Am I right?

Reply:
@jongwu 
Yes, you are right, it's derived from the static variable, that's why I recommended to use a get_pci_root_bus fn to return a static str for different arch just as you did in the go version.

Reply:
@lifupan - what about throwing static variable away as we can declare immutable variable "root_bus_path" using let and initialize it using "create_pci_root_bus_path()" before use it. like :
`let root_bus_path = create_pci_root_bus_path()`

`self.action == U_EVENT_ACTION_ADD
                && self.subsystem == "block"
                && self.devpath.starts_with(&root_bus_path)
                && self.devname != ""
`
So we can keep definition of create_pci_root_bus_path() as :
`
pub fn create_pci_root_bus_path() -> String {
`

Reply:
> @lifupan - what about throwing static variable away as we can declare immutable variable "root_bus_path" using let and initialize it using "create_pci_root_bus_path()" before use it. like :
> `let root_bus_path = create_pci_root_bus_path()`
> 
> `self.action == U_EVENT_ACTION_ADD && self.subsystem == "block" && self.devpath.starts_with(&root_bus_path) && self.devname != "" `
> So we can keep definition of create_pci_root_bus_path() as :
> `pub fn create_pci_root_bus_path() -> String {`

Yeah, that's ok  to me.